cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataProcClient.java

```diff
@@ -222,12 +224,12 @@ public OperationSnapshot createCluster(String name, String imageVersion)
     // figure out the tag to open port 443
     FirewallList firewalls = compute.firewalls().list(conf.getProjectId()).execute();
     List<String> tags = new ArrayList<>();
-    boolean found22 = false;
-    boolean found443 = false;
+    Set<FirewallPort> requiredPorts = EnumSet.allOf(FirewallPort.class);
     for (Firewall firewall : firewalls.getItems()) {
       // network is a url like https://www.googleapis.com/compute/v1/projects/<project>/<region>/networks/<name>
       // we want to get the last section of the path and compare to the configured network name
-      String networkName = Paths.get(URI.create(firewall.getNetwork()).getPath()).getFileName().toString();
+      int idx = firewall.getNetwork().lastIndexOf('/');
+      String networkName = firewall.getNetwork().substring(idx + 1);
```

Comment:
Should check if `idx >=0`. Better be `idx >= 0 ? firewall.getNetwork().substring(idx + 1) : firewall.getNetwork()`.